rust-runtime/aws-smithy-http-server/src/request/lambda.rs

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use lambda_http::request::RequestContext;
+pub use lambda_http::{
+    aws_lambda_events::apigw::{ApiGatewayProxyRequestContext, ApiGatewayV2httpRequestContext},
+    Context,
+};
+
+use crate::Extension;
+
+use super::{extension::MissingExtension, FromParts};
+
+impl<P> FromParts<P> for Context {
+    type Rejection = MissingExtension;
```

Comment:
In each of these cases we're re-using the `MissingExtension` rejection. Do we want rejections tailored specifically to `Context` etc?

Reply:
I'd suggest a little breaking change to `MissingExtension` - allow it to hold a private `'static str` field which we can use to provide a good nudge. For example, in this case, tell the developer to make sure they are using `LambdaHandler`.

Reply:
Will whip up a subsequent PR to make this story coherent across all extractors.